src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -2238,23 +2318,25 @@ def get_storage_account_from_diag_settings(cli_ctx, resource_group_name, name):
     return None
 
 
-def display_diagnostics_report(temp_kubeconfig_path):
+def display_diagnostics_report(temp_kubeconfig_path):   # pylint: disable=too-many-statements
     if not which('kubectl'):
         raise CLIError('Can not find kubectl executable in PATH')
 
-    nodes = subprocess.check_output(["kubectl", "--kubeconfig", temp_kubeconfig_path, "get", "node", "--no-headers"], universal_newlines=True)
+    nodes = subprocess.check_output(
+        ["kubectl", "--kubeconfig", temp_kubeconfig_path, "get", "node", "--no-headers"],
+        universal_newlines=True)
     logger.debug(nodes)
     node_lines = nodes.splitlines()
     ready_nodes = {}
     for node_line in node_lines:
         columns = node_line.split()
         logger.debug(node_line)
         if columns[1] != "Ready":
-            logger.warning("Node {} is not Ready. Current state is: {}.".format(columns[0], columns[1]))
+            logger.warning("Node %s is not Ready. Current state is: %s.", columns[0], columns[1])
         else:
             ready_nodes[columns[0]] = False
 
-    logger.debug('There are {} ready nodes in the cluster'.format(len(ready_nodes)))
```

Comment:
why do you change ".format()" to "%s"?

Reply:
pylint suggests user leverage the functionality of logger.debug/logger.warning/logger.info/logger.fatal.
Instead of construct a string on our own. https://stackoverflow.com/questions/34619790/pylint-message-logging-format-interpolation

Reply:
Got it, thanks for your explanation.

---
In reply to: [358076015](https://github.com/Azure/azure-cli-extensions/pull/1133#discussion_r358076015) [](ancestors = 358076015)